blocks/bricks/__init__.py

```diff
@@ -158,7 +159,56 @@ def _push_initialization_config(self):
                     child.biases_init = self.biases_init
 
 
-class Linear(Initializable):
+class Feedforward(Brick):
+    """Declares an interface for bricks with one input and one output.
+
+    Many bricks have just one input and just one output (activations,
+    :class:`Linear`, :class:`MLP`). To make such bricks interchangable
+    in most contexts they should share an interface for configuring
+    their input and output dimensions. This brick provides such an
+    interface.
+
+    """
+    @abstractproperty
+    def input_dim():
+        """The input dimension of the brick."""
+        pass
+
+    @abstractproperty
+    def output_dim():
+        """The output dimension of the brick."""
+        pass
+
+
+def override_with_attribute(property_, attribute=None):
```

Comment:
Okay, let's keep it simple. I am missing a few features from those stricter languages in Python, in fact, and declaring an interface just by describing the set of class attributes in plain text seems rather weak for me. Because the error message "<something> does not have an attribute input_dim" does not really convey the message that the interface is not respected.

I've just thought that I can override `__getattr__` of `Feedforward` to give a better error message.